src/sinks/aws_s3/sink.rs

```diff
@@ -71,19 +75,27 @@ impl RequestBuilder<(String, Vec<Event>)> for S3RequestOptions {
                 .unwrap_or_else(|| formatted_ts.to_string())
         };
 
+        let ssekms_key_id = metadata.partition_key.ssekms_key_id.clone();
+        let ssekms_key_id_option = (!ssekms_key_id.is_empty()).then_some(ssekms_key_id);
+        let mut s3_options = self.api_options.clone();
+        s3_options.ssekms_key_id = ssekms_key_id_option;
```

Comment:
We then don't need to convert the empty string back into an `Option`.

```suggestion
        let ssekms_key_id = metadata.partition_key.ssekms_key_id.clone();
        let mut s3_options = self.api_options.clone();
        s3_options.ssekms_key_id = ssekms_key_id;
```